domain/consensus/processes/coinbasemanager/coinbasemanager.go

```diff
@@ -182,97 +177,68 @@ func acceptanceDataFromArrayToMap(acceptanceData externalapi.AcceptanceData) map
+	1792387, 1691788, 1596835, 1507212, 1422618, 1342773, 1267409, 1196275, 1129133, 1065759, 1005943, 949484, 896193, 845894, 798417, 753606, 711309, 671386, 633704, 598137, 564566, 532879, 502971, 474742, 448096,
+	422947, 399208, 376803, 355654, 335693, 316852, 299068, 282283, 266439, 251485, 237371, 224048, 211473, 199604, 188401, 177827, 167846, 158426, 149534, 141141, 133219, 125742, 118685, 112024, 105736,
+	99802, 94200, 88913, 83923, 79213, 74767, 70570, 66609, 62871, 59342, 56012, 52868, 49901, 47100, 44456, 41961, 39606, 37383, 35285, 33304, 31435, 29671, 28006, 26434, 24950,
+	23550, 22228, 20980, 19803, 18691, 17642, 16652, 15717, 14835, 14003, 13217, 12475, 11775, 11114, 10490, 9901, 9345, 8821, 8326, 7858, 7417, 7001, 6608, 6237, 5887,
+	5557, 5245, 4950, 4672, 4410, 4163, 3929, 3708, 3500, 3304, 3118, 2943, 2778, 2622, 2475, 2336, 2205, 2081, 1964, 1854, 1750, 1652, 1559, 1471, 1389,
+	1311, 1237, 1168, 1102, 1040, 982, 927, 875, 826, 779, 735, 694, 655, 618, 584, 551, 520, 491, 463, 437, 413, 389, 367, 347, 327,
+	309, 292, 275, 260, 245, 231, 218, 206, 194, 183, 173, 163, 154, 146, 137, 130, 122, 115, 109, 103, 97, 91, 86, 81, 77,
+	73, 68, 65, 61, 57, 54, 51, 48, 45, 43, 40, 38, 36, 34, 32, 30, 28, 27, 25, 24, 22, 21, 20, 19, 18,
+	17, 16, 15, 14, 13, 12, 12, 11, 10, 10, 9, 9, 8, 8, 7, 7, 6, 6, 6, 5, 5, 5, 4, 4, 4,
+	4, 3, 3, 3, 3, 3, 2, 2, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+	0,
 }
 
-func (c *coinbaseManager) calculateSubsidyRandomVariable(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) (int64, error) {
-	ghostdagData, err := c.ghostdagDataStore.Get(c.databaseContext, stagingArea, blockHash, false)
-	if err != nil {
-		return 0, err
+func (c *coinbaseManager) getDeflationaryPeriodBlockSubsidyFromTable(month uint64) uint64 {
+	if month >= uint64(len(subsidyByDeflationaryMonthTable)) {
+		month = uint64(len(subsidyByDeflationaryMonthTable) - 1)
 	}
-	selectedParentHash := ghostdagData.SelectedParent()
-	if selectedParentHash == nil {
-		return 0, nil
-	}
-
-	const binomialSteps = 10
-	binomialSum := int64(0)
+	return subsidyByDeflationaryMonthTable[month]
+}
 
-	// The first two bytes of a hash are a good deterministic source
-	// of randomness, so we use that instead of any rand implementation
-	firstTwoBytes := binary.LittleEndian.Uint16(selectedParentHash.ByteSlice()[:2])
-	for i := 0; i < binomialSteps; i++ {
-		step := firstTwoBytes & 1
-		firstTwoBytes >>= 1
-		binomialSum += int64(step)
-	}
-	return binomialSum - (binomialSteps / 2), nil
+func (c *coinbaseManager) calcDeflationaryPeriodBlockSubsidyFloatCalc(month uint64) uint64 {
+	baseSubsidy := c.deflationaryPhaseBaseSubsidy
+	subsidy := float64(baseSubsidy) / math.Pow(2, float64(month)/12)
+	return uint64(subsidy)
```

Comment:
Can we please not deploy this code as long as it has floats in it?
would appreciate if someone could explain to me the new monetary policy (or link an explanation) so we can come up with a calculation that doesn't involve floats

Reply:
Please read a few lines before as well...

Reply:
In short, we used this method to create the table above, and only the table is actually used